tests/test_fields.py

```diff
@@ -0,0 +1,258 @@
+import sys
+import os
+
+# Sort the paths out to run from this file
+parent_path = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+src_path = os.path.join(os.path.sep,parent_path, 'src')
+sys.path.append(src_path)
+
+from fields import ScalarField, VectorField
+from regressors import CSModel, LModel
+import numpy as np
+import unittest
+
+
+# Key
+# g means grid test (2D field plane)
+# l means line test (1D field line)
+
+
+class Test(unittest.TestCase):
```

Comment:
don't call the class `Test`, it must be more informative, a good system is `TestObject` where `Object` is the name of the object being tested. A given test file can include multiple test classes if you want to test multiple objects from the same module. That way, the test structure matches the code structure, so it's easy to navigate the tests if you know how the code is structured.